updater/lib/dependabot/experimental_grouped_updater.rb

```diff
@@ -0,0 +1,276 @@
+    #        "check" and which parts belong to the "create", but it should be tried
+    #        out in a way that the "dependency" it reports is less of a guess for
+    #        groups -or- we need to implement new error types for Grouped Pull Requests
+    rescue Dependabot::InconsistentRegistryResponse => e
+      log_error(
+        dependency: dependency,
+        error: e,
+        error_type: "inconsistent_registry_response",
+        error_detail: e.message
+      )
+      nil # Return nothing
+    rescue StandardError => e
+      raise if Dependabot::Updater::RUN_HALTING_ERRORS.keys.any? { |err| e.is_a?(err) }
+
+      __getobj__.handle_dependabot_error(error: e, dependency: dependency)
+      nil # Return nothing
+    end
+    # rubocop:enable Metrics/MethodLength
+
+    def filter_unrelated_and_unchanged(updated_dependencies, checker)
+      updated_dependencies.reject do |d|
+        next false if d.name == checker.dependency.name
+        next true if d.top_level? && d.requirements == d.previous_requirements
+
+        d.version == d.previous_version
+      end
+    end
+
+    def update_files_and_create_pull_request(update_batch)
+      # FIXME: This is a workround for not having a single Dependency to report against
+      #
+      #        We could use update_batch.values.first, but it might make more
+      #        sense to report dependency || group for the error types involved
+      #        so the problem is correctly attributed to the update creation
+      #        as opposed to checking any one dependency.
+      #
+      #        Alternatively, when we use a shared workspace to build the diff iteratively
+      #        we could report against the _active_ dependency in each step. At worst this
+      #        might result in an off-by-one issue if we have potential for an error between
+      #        loops.
+      group_dependency = OpenStruct.new(name: "group-all")
+
+      logger_info("Generated #{update_batch.count} changes")
+      # FIXME: This needs to be de-duped, but it isn't clear which dupe should
+      #        'win' right now in terms of version.
+      #
+      #        To start out with, using a variant on the 'existing_pull_request'
+      #        logic might make sense -or- we could employ a one-and-done rule
+      #        where the first update to a dependency blocks subsequent changes.
+      #
+      #        In a follow-up iteration, a 'shared workspace' could provide the
+      #        filtering for us assuming we iteratively make file changes for
+      #        each Array of dependencies in the batch and the FileUpdater tells
+      #        us which cannot be applied.
+      all_updated_deps = update_batch.values.flatten
+      logger_info("Updated #{all_updated_deps.count} dependencies")
+      updated_files = generate_dependency_files_for(all_updated_deps)
+
+      create_pull_request(all_updated_deps, updated_files, pr_message(all_updated_deps, updated_files))
```

Comment:
This passes existing tests but is too naive to work under most circumstances, this is where we need to replicate the iterative file changes from the [dry-run prototype](https://github.com/dependabot/dependabot-core/pull/5867/files#diff-095eccc30d572b6b06c61c5b23d291ddcdaca2b3255650fd303590a86781d964R780-R788).

@bdragon I think this is essentially where the control plane for the underlying file system would be, with `generate_dependency_files_for` being executed per row of `update_batch.values` and the outcome resulting in either a `git commit` or `git clean`.